dao/src/main/java/greencity/repository/UserRepo.java

```diff
@@ -220,9 +223,19 @@
     /**
      * Find and return all registration months.
      *
-     * @return {@link List} of months
+     * @return {@link List} of {@link RegistrationStatisticsDtoResponse}
      **/
-    @Query("SELECT month(u.dateOfRegistration) - 1 FROM User u WHERE "
-        + "year(u.dateOfRegistration) = year(CURRENT_TIMESTAMP)")
-    List<Integer> findAllRegistrationMonths();
+    @Query(nativeQuery = true)
+    List<RegistrationStatisticsDtoResponse> findAllRegistrationMonths();
+
+    /**
+     * Converts result of findAllRegistrationMonths() method to {@link Map}.
+     *
+     * @return {@link Map}
+     */
+    default Map<Integer, Long> findAllRegistrationMonthsMap() {
+        return findAllRegistrationMonths().stream().collect(Collectors.groupingBy(
```

Comment:
Why do you need grouping?
I think, you already have 12 unique elements in the query result. You just need to convert it into a map, for example

`return findAllRegistrationMonths().stream().collect(Collectors.toMap(RegistrationStatisticsDtoResponse::getMonth, RegistrationStatisticsDtoResponse::getCount));`

Reply:
Yeah, did it by mistake.